src/pinning.js

```diff
@@ -118,16 +119,24 @@ class Pinning {
       this.orbitdb._onMessage = messageBroker.onMessageWrap.bind(messageBroker)
     }
     this.pubsub = new Pubsub(this.ipfs, ipfsId.id)
-    this.pubsub.subscribe(this.pinningRoom, this._onMessage.bind(this), this._onNewPeer.bind(this))
-    setInterval(this.checkAndCloseDBs.bind(this), this.dbCheckCloseInterval)
+    await this.pubsub.subscribe(this.pinningRoom, this._onMessage.bind(this), this._onNewPeer.bind(this))
+    this._intervals.push(setInterval(this.checkAndCloseDBs.bind(this), this.dbCheckCloseInterval))
   }
 
-  checkAndCloseDBs () {
-    Object.keys(this.openDBs).map(async key => {
+  async stop () {
+    await this.pubsub.disconnect()
+    await this.checkAndCloseDBs()
+    await this.orbitdb.stop()
+    await this.ipfs.stop()
+    this._intervals.forEach(clearInterval)
```

Comment:
nice, more thoughtful cleanup, does this._intervals need to an array?

Reply:
No, I thought it would be a good way of reminding us to track async calls and shut the down correctly, but maybe it adds confusion

Reply:
definitely like having clearInterval, just didnt follow at first since we only have one case of that and this was generalized, but yeah good either way